torchtune/utils/_distributed.py

```diff
@@ -297,3 +299,31 @@ def lora_wrap_fsdp(module: nn.Module, recurse: bool, **kwargs):
         return isinstance(module, tuple(modules_to_wrap))
 
     return lora_wrap_fsdp
+
+
```

Comment:
@awgu for FSDP checkpoint util, should we put it here in torchtune or move it into pytorch? it seems to be needed in general beyond torchtune

Reply:
My current personal opinion: We should not land it in `pytorch` repo yet.
1. We do not know the problem space well enough to know if the generality of this utility and how users may want to customize it (e.g. add coalescing of broadcasts, add quantization, etc.).
2. The utility does not have a high degree of complexity. The building block doing the heavy lifting is `DTensor`. This function can be copy-and-pasted and tailored to specific needs.

We can see how the torchtune folks feel about having it in their utils folder for now.